elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -413,23 +413,30 @@ export class PeerPool extends EventEmitter {
 			disconnectedFixedPeers.length -
 			outboundCount;
 
+		const bestPeersbyHeight = getBestPeersbyHeight([
+			...disconnectedNewPeers,
+			...disconnectedTriedPeers,
+			...disconnectedFixedPeers,
+		]);
+
 		// This function can be customized so we should pass as much info as possible.
 		const peersToConnect = this._peerSelectForConnection({
-			newPeers: disconnectedNewPeers,
-			triedPeers: disconnectedTriedPeers,
+			newPeers: comparePeerListsByHeight(
```

Comment:
we don't this here, we can simply filter out the duplicates in the end while making an inbound